build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -842,6 +842,58 @@ message NodeProperties {
 
   // The UNIX file mode, e.g., 0755.
   google.protobuf.UInt32Value unix_mode = 3;
+
+  // Constrains server-side unification of identical filesystem nodes.
+  NodeUnificationConstraint.Value unification_constraint = 4;
+}
+
+// A `NodeUnificationConstraint` specifies whether servers are allowed to 
```

Comment:
How would you feel about flipping this? Instead of mixing positive ("ALLOW_X") and negatives ("FORBID_X"), only stick to negatives? If we have the default be that the server is generally free to optimize, then this field would be for adding one or more restrictions against what the server is allowed to do.

My first worry is that having clients opt-_in_ to optimizations seems like it'll lead to clients missing viable optimizations commonly, and a pile of "allow_X" optimizations on every single node from the rest. That also seems like it may be more future-proof: if we add a new optimization field, does every existing client not explicitly allowing it effectively stop being eligible for that optimization? Whereas we'd only start adding new ways to forbid optimizations if/when they were found to matter for a few cases. (If they only ever _apply_ in a few cases, I guess I could see this, or just specifying the recommendation to do something in a slightly different way.)